festim/exports/average_surface.py

```diff
@@ -1,32 +1,35 @@
-from dolfinx import fem
 import festim as F
-import csv
 import numpy as np
 
 
 class AverageSurface(F.SurfaceQuantity):
-    """
+    """Computes the average value of a field on a given surface
+
     Args:
-        field (str): the field from which the average value is computed
-        surface (festim.SurfaceSubdomain1D): the surface id where the average value is computed
-        filename (str, optional): name of the file to which the total value is computed
-    """
+        field (festim.Species): species for which the average surface is computed
+        surface (festim.SurfaceSubdomain): surface subdomain
+        filename (str, optional): name of the file to which the average surface is exported
 
-    def __init__(self, field, surface, filename: str = None) -> None:
-        super().__init__(field, surface, filename)
-        self.function = function
+    Attributes:
+        see `festim.SurfaceQuantity`
+    """
 
+    def __init__(
+        self,
+        field: F.Species,
+        surface: F.SurfaceSubdomain,
+        filename: str = None,
+    ) -> None:
+        super().__init__(field=field, surface=surface, filename=filename)
+    
     @property
-    def function(self):
-        return self.function
-
-    @function.setter
-    def function(self, funct):
-        if not isinstance(funct, (dolfinx.fem.function.Function)):
-            raise TypeError("function must be of type dolfinx.fem.function.Function")
-
-        self._function = funct
+    def title(self):
+        return f"Average {self.field.name} surface {self.surface.id}"
 
     def compute(self):
-        """Compute the total value of the function on the surface"""
-        # placeholder for the coming function
+        """
+        Computes the average value of the field on the defined surface
+        subdomain, and appends it to the data list
+        """
+        self.value = np.mean(self.field.solution.x.array[self.surface.indices])
```

Comment:
No this won't work.

Imagine you have an array of 100 cells with 90 that are at value 10.
The other 10 cells have a value 1.

The average according to your implementation is 910/100=9.10

But you have no information about the localisation of these cells.

Maybe the 90 cells are only located on a very small region and 85% of the domain is at value 1?

The correct formulation for the average of a function f on a surface is:

integral(f * dS)/integral(1 * dS)